honeybee/radiance/recipe/daylightfactor/gridbased.py

```diff
@@ -30,6 +32,29 @@ def __init__(self, analysis_grids, rad_parameters=None, hb_objects=None,
             self, sky, analysis_grids, simulation_type, rad_parameters, hb_objects,
             sub_folder)
 
+    @classmethod
+    def from_json(cls, rec_json):
+        """Create Daylight Factor recipe from json.
+            {
+            "id": 2, // do Not overwrite this id
+            "analysis_grids": [], // list of analysis grids
+            "surfaces": [], // list of honeybee surfaces
+            "rad_parameters": {
+                gridbased_parameters: string //  A standard radiance parameter string
+                (e.g. -ab 5 -aa 0.05 -ar 128)
+                },
+            }
+        """
+        analysis_grids = \
+            tuple(AnalysisGrid.from_json(ag) for ag in rec_json["analysis_grids"])
+        hb_objects = tuple(HBSurface.from_json(srf) for srf in rec_json["surfaces"])
+        rad_parameters = GridBasedParameters.from_json(rec_json["rad_parameters"])
+
+        recipe = cls(analysis_grids = analysis_grids, rad_parameters=rad_parameters, \
+        hb_objects=hb_objects)
```

Comment:
next line should be 4 white spaces ahead of the line before.

Reply:
Not sure what you mean here. Have corrected as I have understood your comment.